agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -0,0 +1,617 @@
+	if !ok {
+		return "", errors.New("unable to obtain credentialspec mapping")
+	}
+
+	return targetCredSpecMapping, nil
+}
+
+func (cs *CredentialSpecResource) updateCredSpecMapping(credSpecInput, targetCredSpec string) {
+	cs.lock.Lock()
+	defer cs.lock.Unlock()
+
+	cs.CredSpecMap[credSpecInput] = targetCredSpec
+}
+
+// Cleanup removes the credentialspec created for the task
+func (cs *CredentialSpecResource) Cleanup() error {
+	cs.clearCredentialSpec()
+	return nil
+}
+
+// clearCredentialSpec cycles through the collection of credentialspec data and
+// removes them from the task
+func (cs *CredentialSpecResource) clearCredentialSpec() {
+	cs.lock.Lock()
+	defer cs.lock.Unlock()
+
+	for key, value := range cs.CredSpecMap {
+		if strings.HasPrefix(key, "credentialspec:file://") {
+			seelog.Debugf("Skipping cleanup of local credentialspec file option: %s", key)
+			continue
+		}
+		credSpecSplit := strings.SplitAfterN(value, "credentialspec=", 2)
+		localCredentialSpecFile := credSpecSplit[1]
+		err := cs.os.Remove(localCredentialSpecFile)
+		if err != nil {
+			seelog.Debugf("Unable to clear local credential spec file %s for task %s", localCredentialSpecFile, cs.taskARN)
+		}
+		delete(cs.CredSpecMap, key)
+	}
+}
+
+// CredentialSpecResourceJSON is the json representation of the credentialspec resource
+type CredentialSpecResourceJSON struct {
+	TaskARN                 string                               `json:"taskARN"`
+	CreatedAt               *time.Time                           `json:"createdAt,omitempty"`
+	DesiredStatus           *CredentialSpecStatus                `json:"desiredStatus"`
+	KnownStatus             *CredentialSpecStatus                `json:"knownStatus"`
+	RequiredCredentialSpecs map[string][]*apicontainer.Container `json:"credentialSpecResources"`
+	CredSpecMap             map[string]string                    `json:"CredSpecMap"`
+	ExecutionCredentialsID  string                               `json:"executionCredentialsID"`
+}
+
+// MarshalJSON serialises the CredentialSpecResourceJSON struct to JSON
+func (cs *CredentialSpecResource) MarshalJSON() ([]byte, error) {
+	if cs == nil {
+		return nil, errors.New("credential specresource is nil")
+	}
+	createdAt := cs.GetCreatedAt()
+	return json.Marshal(CredentialSpecResourceJSON{
+		TaskARN:   cs.taskARN,
+		CreatedAt: &createdAt,
+		DesiredStatus: func() *CredentialSpecStatus {
+			desiredState := cs.GetDesiredStatus()
+			s := CredentialSpecStatus(desiredState)
+			return &s
+		}(),
+		KnownStatus: func() *CredentialSpecStatus {
+			knownState := cs.GetKnownStatus()
+			s := CredentialSpecStatus(knownState)
+			return &s
+		}(),
+		RequiredCredentialSpecs: cs.getRequiredCredentialSpecs(),
+		CredSpecMap:             cs.getCredSpecMap(),
+		ExecutionCredentialsID:  cs.getExecutionCredentialsID(),
+	})
+}
+
+// UnmarshalJSON deserialises the raw JSON to a CredentialSpecResourceJSON struct
+func (cs *CredentialSpecResource) UnmarshalJSON(b []byte) error {
```

Comment:
Why unmarshalJSON does not have CredSpecMap field?

Reply:
Good catch, thanks! Fixed it now.